modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java

```diff
@@ -218,6 +224,35 @@ protected void doServeResource(
 		}
 	}
 
+	private JSONArray _getAllVersionsJSONArray(
+		ContentDashboardItem contentDashboardItem, ThemeDisplay themeDisplay) {
+
+		int status = WorkflowConstants.STATUS_APPROVED;
+
+		PermissionChecker permissionChecker =
+			themeDisplay.getPermissionChecker();
+
+		User user = themeDisplay.getUser();
+
+		if ((user.getUserId() == contentDashboardItem.getUserId()) ||
```

Comment:
Hey @jkappler !!

What do you think about moving that logic to the ContentDashboardItem, so every implementation can decide what are the status of the versions they should return?

Reply:
Hi @cgoncas, will change this

Reply:
Updated PR with the changes, every ContentDashboardItem will decided what to return